pulp_smash/tests/pulp3/pulpcore/api_v3/test_tasks.py

```diff
@@ -0,0 +1,94 @@
+# coding=utf-8
+"""Test that operations can be performed over tasks."""
+import unittest
+from urllib.parse import urljoin
+
+from requests import HTTPError
+
+from pulp_smash import api, config, selectors, utils
+from pulp_smash.tests.pulp3.constants import REPO_PATH, TASKS_PATH
+from pulp_smash.tests.pulp3.pulpcore.utils import gen_repo
+from pulp_smash.tests.pulp3.utils import get_auth
+from pulp_smash.tests.pulp3.pulpcore.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+
+_DYNAMIC_TASKS_ATTRS = ('finished_at',)
+"""Task attributes that are dynamically set by Pulp, not set by a user."""
+
+
+class TasksTestCase(unittest.TestCase, utils.SmokeTest):
+    """Perform different operation over tasks.
+
+    This test targets the following issues:
+
+    * `Pulp #3144 <https://pulp.plan.io/issues/3144>`_
+    * `Pulp #3527 <https://pulp.plan.io/issues/3527>`_
+    * `Pulp Smash #754 <https://github.com/PulpQE/pulp-smash/issues/754>`_
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        cls.client = api.Client(cls.cfg, api.json_handler)
+        cls.client.request_kwargs['auth'] = get_auth()
+        cls.task = {}
+
+    def test_01_create_task(self):
+        """Create a task."""
+        repo = self.client.post(REPO_PATH, gen_repo())
+        self.addCleanup(self.client.delete, repo['_href'])
+        attrs = {'description': utils.uuid4()}
+        response = self.client.patch(repo['_href'], attrs)
+        self.task.update(self.client.get(response['_href']))
+
+    @selectors.skip_if(bool, 'task', False)
+    def test_02_read_task(self):
+        """Read a task by its _href."""
+        task = self.client.get(self.task['_href'])
+        for key, val in self.task.items():
+            if key in _DYNAMIC_TASKS_ATTRS:
+                continue
+            with self.subTest(key=key):
+                self.assertEqual(task[key], val)
+
+    @selectors.skip_if(bool, 'task', False)
+    def test_03_read_tasks(self):
+        """Read a task by its worker."""
+        with self.subTest('invalid worker name'):
+            with self.assertRaises(HTTPError):
+                self.filter_tasks({'worker': utils.uuid4()})
+        with self.subTest('valid worker name'):
+            page = self.filter_tasks({'worker': self.task['worker']})
+            self.assertNotEqual(len(page['results']), 0)
+
+    def test_04_filter_invalid_date(self):
```

Comment:
How about a complementary test method that filters by a valid date?

Reply:
I tried it before, but tests that involve time is hard to foreseen results. I will try to come up with another test to accomplish this.

Reply:
The task created earlier in this test case should work, no?